pyugrid/read_netcdf.py

```diff
@@ -150,9 +150,11 @@ def load_grid_from_nc_dataset(nc, grid, mesh_name=None, load_data=True):
                 raise ValueError(msg(defs['role']))
             continue
         except KeyError:
-            msg = ("File must include {} variables for {} "
-                   "named in mesh variable.").format
-            raise ValueError(msg(coord_names, defs['role']))
+            if defs['required']:
+                msg = ("File must include {} variables for {} "
+                       "named in mesh variable.").format
+                raise ValueError(msg(coord_names, defs['role']))
+            continue
```

Comment:
I think it would be clearer if each line of

```python
coord_names = mesh_var.getncattr(defs['role']).strip().split()
coord_vars = [nc.variables[name] for name in coord_names]
```

had its own `try`/`except` clause.

In the end this is compliance checking, right?  It would be nice to refactor this, in the long run of course, to perform the compliance checking before creating the ugrid object and to expose the compliance checking API.